miner/scroll_worker.go

```diff
@@ -845,6 +845,20 @@ func (w *worker) commit(force bool) (common.Hash, error) {
 		)
 	}
 
+	ancestorHeight := uint64(1)
+	currentHeight := w.current.header.Number.Uint64()
+	maxReorgDepth := uint64(w.config.CCCMaxWorkers + 1)
+	if currentHeight > maxReorgDepth {
+		ancestorHeight = currentHeight - maxReorgDepth
+	}
+
+	ancestorHash := w.chain.GetHeaderByNumber(ancestorHeight).Hash()
+	if rawdb.ReadBlockRowConsumption(w.chain.Database(), ancestorHash) == nil {
+		// reject committing to a block if its ancestor doesn't have its RC stored in DB yet.
+		// which may either mean that it failed CCC or it is still in the process of being checked
+		return common.Hash{}, retryableCommitError{inner: errors.New("ancestor doesn't have RC yet")}
```

Comment:
Should we also print a warn log here? Or do you expect that the latter case might be common?

Reply:
I was expecting caller to print it, but just realized that we don't log retryable errors. added in https://github.com/scroll-tech/go-ethereum/pull/1022/commits/2b77324dbba27a982d17690e5679760ccb24e5b1